virtcontainers/container.go

```diff
@@ -405,6 +405,14 @@ func (c *Container) unmountHostMounts() error {
 					"error":     err,
 				}).Warn("Could not umount")
 				return err
+			} else if err := os.RemoveAll(m.HostPath); err != nil {
```

Comment:
I think you should be able to create a unit test for this in `container_test.go` where you set `c.mounts[0]` to contain a double bind-mounted path. That way, the `Unmount()` will succeed but the `RemoveAll()` will fail (as there is still one more `Unmount()` call to make).